SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUpstreamPoller.java

```diff
@@ -0,0 +1,32 @@
+package com.hubspot.singularity.scheduler;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.hubspot.singularity.config.SingularityConfiguration;
+
+@Singleton
+public class SingularityUpstreamPoller extends SingularityLeaderOnlyPoller {
+  private static final Logger LOG = LoggerFactory.getLogger(SingularityUpstreamPoller.class);
+
+  private final SingularityUpstreamChecker upstreamChecker;
+
+  @Inject
+  SingularityUpstreamPoller(SingularityConfiguration configuration, SingularityUpstreamChecker upstreamChecker) {
+    super(configuration.getCheckUpstreamsEverySeconds(), TimeUnit.SECONDS, true);
+
+    this.upstreamChecker = upstreamChecker;
+
+  }
+
+  @Override
+  public void runActionOnPoll() {
+    LOG.info("Checking upstreams");
```

Comment:
Can we add in a configuration param for whether or not this poller is enabled and short circuit here if needed? Will make it much easier to turn off the functionality in case something ever goes wrong with it. (vs rolling back builds/etc)